usr/share/rear/restore/COVE/default/400_restore_with_cove.sh

```diff
@@ -0,0 +1,370 @@
+        # Create symlinks to redirect the installation to a disk
+        if [ "$COVE_INSTALL_DIR" != "$COVE_REAL_INSTALL_DIR" ]; then
+            local target="$target_install_dir"
+            local link_name="$COVE_INSTALL_DIR"
+            cove_create_symlink "${target}" "${link_name}" || return $?
+        else
+            cove_dirs=(bin etc lib sbin share temp var/log var/storage)
+            for cove_dir in "${cove_dirs[@]}"; do
+                local target="${target_install_dir}/${cove_dir}"
+                local link_name="${COVE_INSTALL_DIR}/${cove_dir}"
+                cove_create_symlink "$target" "$link_name" || return $?
+            done
+        fi
+    fi
+
+    UserOutput ""
+    UserOutput "Installing Backup Manager..."
+    "$COVE_INSTALLER_PATH" --target "${COVE_TMPDIR}/mxb" 1>&7 2>&8 || return $?
+
+    # Extract the ReaR tarball because the installer does not do it in the rescue environment
+    mkdir -p "${target_install_dir}/rear"
+    tar -xf "${COVE_TMPDIR}/mxb/rear.tar.gz" --strip-components=1 -C "${target_install_dir}/rear"
+
+    # Try to copy site.conf if it does not exist at the target.
+    # It happens in case of symlinks used to redirect the installation.
+    [ -e "${target_install_dir}/rear/etc/rear/site.conf" ] \
+        || cp "${COVE_INSTALL_DIR}/rear/etc/rear/site.conf" "${target_install_dir}/rear/etc/rear/site.conf" \
+        || true
+}
+
+# Stops ProcessController process
+function cove_stop_pc() {
+    local pid
+    pid="$(ps aux | awk -v pc_name=ProcessController '$0 ~ pc_name && !/awk/ {print $2}')"
+    [ -z "$pid" ] || { /bin/kill -TERM "$pid" && \
+    while [ -n "$pid" ]; do \
+        sleep 1; \
+        pid="$(get_pc_pid)"; \
+    done }
+}
+
+# Print the welcome message
+UserOutput "
+The System is now ready for restore. The Backup Manager installer will be
+downloaded and run automatically. If any required parameters have not been
+provided, you will be prompted to enter them."
+
+# Read parameters from boot options. The existing values can be overridden
+# by values passed via boot options.
+read -r cmdline </proc/cmdline
+for option in $cmdline; do
+    case $option in
+        (cove_installer=*)
+            COVE_INSTALLER_URL="${option#cove_installer=}"
+            ;;
+        (cove_token=*)
+            COVE_INSTALLATION_TOKEN="${option#cove_token=}"
+            ;;
+        (cove_timestamp=*)
+            COVE_TIMESTAMP="${option#cove_timestamp=}"
+            ;;
+    esac
+done
+
+COVE_TMPDIR="$(mktemp -d "${TARGET_FS_ROOT}/cove.XXXXXXXXXX")"
+readonly COVE_TMPDIR
+
+COVE_INSTALLER_PATH="${COVE_TMPDIR}/mxb-linux-x86_64.run"
+
+prompt="Select what to do"
+rear_workflow="rear $WORKFLOW"
+
+unset choices
+choices[0]="Try downloading the Backup Manager installer again"
+choices[1]="Change the Backup Manager installer URL"
+choices[2]="Use Relax-and-Recover shell and return back to here"
+choices[3]="Abort '$rear_workflow'"
+choice=""
+
+while true; do
```

Comment:
This and the following seems like an inverted control ReaR vs. COVE. Normally ReaR setups up everything, starts the backup client and triggers a restore (or, if the backup software doesn't support client initiated operations it asks the admin to go and run the restore). ReaR then monitors the restore status and continues with the recovery process once that is done.